homeassistant/components/switch/template.py

```diff
@@ -0,0 +1,170 @@
+
+        if device != slugify(device):
+            _LOGGER.error("Found invalid key for switch.template: %s. "
+                          "Use %s instead", device, slugify(device))
+            continue
+
+        if not isinstance(device_config, dict):
+            _LOGGER.error("Missing configuration data for switch %s", device)
+            continue
+
+        friendly_name = device_config.get(ATTR_FRIENDLY_NAME, device)
+        state_template = device_config.get(CONF_VALUE_TEMPLATE)
+        on_action = device_config.get(ON_ACTION)
+        off_action = device_config.get(OFF_ACTION)
+        if state_template is None:
+            _LOGGER.error(
+                "Missing %s for switch %s", CONF_VALUE_TEMPLATE, device)
+            continue
+
+        if on_action is None or off_action is None:
+            _LOGGER.error(
+                "Missing action for switch %s", device)
+            continue
+
+        switches.append(
+            SwitchTemplate(
+                hass,
+                device,
+                friendly_name,
+                state_template,
+                on_action,
+                off_action)
+            )
+    if not switches:
+        _LOGGER.error("No switches added")
+        return False
+    add_devices(switches)
+    return True
+
+
+class SwitchTemplate(SwitchDevice):
+    """ Represents a Template Switch. """
+
+    # pylint: disable=too-many-arguments
+    def __init__(self,
+                 hass,
+                 device_id,
+                 friendly_name,
+                 state_template,
+                 on_action,
+                 off_action):
+
+        self.entity_id = generate_entity_id(
+            ENTITY_ID_FORMAT, device_id,
+            hass=hass)
+
+        self.hass = hass
+        self._name = friendly_name
+        self._template = state_template
+        self._on_action = on_action
+        self._off_action = off_action
+        self.update()
+
+        def _update_callback(_event):
+            """ Called when the target device changes state. """
+            # This can be called before the entity is properly
+            # initialised, so check before updating state,
+            if self.entity_id:
+                self.update_ha_state(True)
+
+        self.hass.bus.listen(EVENT_STATE_CHANGED, _update_callback)
+
+    @property
+    def name(self):
+        """ Returns the name of the device. """
+        return self._name
+
+    @property
+    def should_poll(self):
+        """ Tells Home Assistant not to poll this entity. """
+        return False
+
+    def turn_on(self, **kwargs):
+        """ Fires the on action. """
+        call_from_config(self.hass, self._on_action, True)
+
+    def turn_off(self, **kwargs):
+        """ Fires the off action. """
+        call_from_config(self.hass, self._off_action, True)
+
+    @property
+    def is_on(self):
+        """ True if device is on. """
+        return self._state == STATE_TRUE or self._state == STATE_ON
```

Comment:
I'm using 'True' is to match what the template generates when you do a boolean comparison.

Of course it can also generate 'error'. 
![screen shot 2016-02-04 at 17 12 10](https://cloud.githubusercontent.com/assets/4487313/12823164/780df722-cb62-11e5-91ca-79a220c57133.png)

My thought was that as a user you'd expect to be able to produce a bool value in the template and have it interpreted correctly.